### DISCUSSION TITLE
delay on own audio signal with windows + Behringer UMC204HD

### DISCUSSION THREAD

**Discussion by sbg-1** [jamulussoftware/jamulus#3427] [Technical Support & Help]
Hi there,

I am new to Jamulus, and joined in order to continue my (acoustic guitar) lessons with my guitar teacher, who has moved across country but continues to teach online through Jamulus.

I purchased a Behringer U-Phoria UMC204HD audio interface in order to connect my mic to jamulus. I specifically chose this interface because another student of hers has been using it successfully for the same purpose. For me, however, I can't seem to get it to work properly. I feel like I’ve tried everything I can think of, I'm now not sure what to do anymore, and hoping someone here knows what else I can try.

I will try to outline what the problem is, what system I have, and what I have already tried to fix the issue below but it is a bit of a “wall of text” for which I apologize in advance. 

**The problem:** 
The problem I am experiencing is that there is significant delay on my own audio signal, making it difficult to talk and almost impossible to play guitar (it almost sounds like a flanger effect). My guitar teacher gets my signal with no to minimal delay and clear, and can keep in sync with me just fine. I hear her just fine as well, with minimal delay, but curiously, I hear my own signal with significant delay. Her other students don’t experience this. I’ve actually gone over to another student’s house (who also uses the Behringer UMC204HD but with a Mac, more on that later) and he does not have the delay on his own signal at all. It was perfectly possible to play guitar listening to my own signal in my headphones. 

**My setup is as follows:** 
- A Dell XPS laptop running windows 10 
- A Rode NT1 (4th gen) microphone, used to pick up both my voice and my acoustic guitar (same as the other students)
- Sennheiser HD 280 pro headphones
- (stable and fast) internet connection to the router via an ethernet cable (+usb adapter piece) 
-the Behringer UMC204HD, plus the driver version 5.57, which shows up in Jamulus just fine (as “UMC ASIO driver”). 
-my guitar teacher has a dedicated server, and, as mentioned, successfully teaches other students this way. 
-(I tried to figure out which version of jamulus I have, but it doesn't show it under properties. i downloaded it from the jamulus website around the 20th of september though, so it should be newest one?)

**What I have already tried:** 
-I've played with all the settings on Jamulus that I could think of, but that doesn't really seem to do anything. 
-I also checked my laptop’s settings,  specifically that the Behringer audio interface is set as the correct input and output device and on (what seem to me) the appropriate channel(s).
-I matched all the settings of the other student with the Behringer’s UMC204HD interface on my own interface. 

**Furthermore, I checked the following**:
-It's not a problem with the hardware itself, because I tested the headphones, microphone and the Behringer interface all separately with the other student’s (working) setup. He himself has does have a Mac desktop (and thus doesn’t have to download the behringer driver separately), and there it all runs perfectly fine with my interface + mic + headphones. As soon as I hooked it up to my laptop again, however, I got the same delay issue, even on his network (so that doesn't seem to be it either). 
-Just out of curiosity I then tried another ASIO driver (ASIO4ALL) that happened to still be on my laptop (from a different mic set up that also did not work, I've been at this for a while now). To my surprise, the delay was almost gone and the issue was basically “fixed”. (Unfortunately, this driver also controls the internal boxes and microphone on my laptop, and there is not really a way I can find to turn those off which causes interference between the Rode mic and the internal mic. So continuing to use that isn't really an option.) 

The conclusion I’ve drawn from this weeks long troubleshooting session is that the problem must be somewhere between my laptop and the Behringer driver. Perhaps it was a problem with windows 10 and my outdated laptop? 
-So, just in case it might be a windows 10/my laptop problem, I also tried the same setup on another laptop running windows 11 (a Dell inspiron 15 3515), but there again I had the exact same problem: an unworkable delay on my own audio signal. (To really deepen the mystery/my befuddlement, I do know of someone who uses a windows 11 laptop with a UMC404HD and Jamulus just fine… so it is possible on windows just not for me???) 

With that, I've basically exhausted all the options I can think of, and I don't know what to try anymore to get the Behringer interface to work. Admittedly I'm not much of an audio/tech girlie, but I feel like I've tried everything I can think of already.
-(I am still going to try my guitar teacher's audio interface, who has a Focusrite scarlett 4th gen running on a windows 11 PC. Maybe that one (+ its driver) will work, and then I'll know for sure that the problem really lies with Behringer's driver+windows/my pc. But it will be a while before I can meet up with her, and if I decide to return the Behringer interface to the dealer because I cannot get it to work, I will have to do so in accordance with the dealer’s return policy before I can try my teacher’s scarlett interface).

I don't know what more I can do to get Jamulus+ the behringer interface working properly on my PC. The fact that it does work perfectly with a Mac, but not for me on another PC with windows 11 baffles me, and I’m kind of at my wit’s end with this problem. Surely one would think that if there actually is a problem with Behringer's driver for both windows 10 and 11 that a simple google search would find many more people struggling with this? And yet, I don’t see that pop up either when I do. 

I have contacted Behringer tech support and have yet to hear from them, but figured that maybe someone versed in Jamulus might also have some useful input. 

If you’ve made it to here, thank you so much for reading through all of this and I hope someone can help me fix this issue and continue my guitar lessons! Or least give me some new things to try before I have to return the Behringer :)

**Comment by jujudusud** [comment]
How did you set the frame size in UMC control panel?
Is the headphone directly connected to the USB interface?
Did you uninstall asio4all properly?

**Reply by sbg-1** [reply]
Thanks for your replies!

1) I don't know if I can set the frame size in the UMC control panel tbh? When I open it I get options for sample rate (48000Hz) and preferred ASIO buffer size (182 samples). I don't see anything about frame size. A quick google search didn't provide more clarity on this issue either. Am I missing something?

2) yes, the headphones are directly plugged into the audio interface

3) no, i did not uninstall asio4all yet. That's how I figured out (by accident) that it is possible to listen to my own sound without (noticeable) delay, by selecting asio4all as my driver in jamulus (the only problem with that is that it also drives my laptop's mic and speakers, so feedback becomes an issue). Do you think uninstalling it would make any difference? 

someone else also asked if I could swap my interface with the other students one and test if it's a device issue (but curiously that message doesn't show up here as it did in my email)
4) no, I haven't tried that. I did connect my interface to his Macbook, and there it worked just fine so I didn't think to connect his interface to my laptop. Because mine worked just fine with his mac i'm guessing it isn't a device issue, but I can always try his and check for sure.

Thanks again for your input!

**Reply by jujudusud** [reply]
1. Sample rate at 48kHz IS ok. You can go down to 128 samples for the buffer size and disable the "safe mode" radio button. No you miss nothing at all. ;-)
2. Nice.
3. The advice in Jamulus' manual is : Use the customers ASIO driver if it is existing. You should uninstall ASIO4ALL and Don't use it With your Behringer UMC.
4. Look at the interface button "input/playback" ans put it fully to playback side.

**Reply by sbg-1** [reply]
Hi!
Thanks for getting back to me and your suggestions! Sorry for the belated response.

Here are my findings: 
1. OK, I did, nothing much changed though. But nice to know I didn't miss anything :-D
2. Nice.
3. I uninstalled all the other drivers, that didn't do much either*
4. I changed this (on in interface itself I presume), also didn't seem to make much of a difference.
5. (in complete desperation) I also tried a different hub to connect my laptop to the lan cable (my laptop doesn't have a dedicated port), and that shaved off a few seconds on Jamulus' ping/delay display, but nothing noticeably changed in the delay on my own audio signal
6. I've also been in contact with Behringer support, they suggested uninstalling the newer driver for the interface and installing an older version for troubleshooting purposes, and that didn't change anything either. the signal was more or less the same. I've reported this back to them and will wait for their response.
7. I'll have access to two different audio interfaces somewhere in the coming two weeks (a yamaha and a focusrite) so I'll have to wait on those to see if I have the same issues or if its a behringer specific issue. 

*before I uninstalled it, I did test out the asio4all driver with my guitar teacher on the other end. When I previously tested it on her empty server my audio seemed noticeably better (less delay) so I figured that meant that the issue was with the behringer driver. However, when I tried switching to asio4all with my teacher on the other end I didn't hear anything at all anymore and she heard me much worse. I'm not entirely sure what this means (other than that I have no need to reinstall asio4all ;-), but it does maybe lessen the "evidence" that it's definitely the behringer driver that's causing the issue... maybe? I'm not sure of anything anymore other than that my head hurts from thinking about this so much :'-)

Still very much open to suggestions, and I will also update here if I find out anything new. Thanks!

**Comment by QJKX** [comment]
Once connected, what are the _Ping_ and _Delay_ numbers?

**Reply by sbg-1** [reply]
Hi! Thanks for your response and sorry for my late reply! I was traveling and couldn't check until now. 

To answer your question, it hops around a lot. _ping_ is between 22-45, and averages (i guess) around 30. _delay_ is between 80-120, averaging around 85? This is without skype connection, which we also use during lessons. I'm sure this drives those numbers up, but I can't check that right now. If that is useful to know lemme know and I can check next wednesday, when my lesson is planned. 

Do these numbers tell you anything? 

Thanks again!

**Reply by ann0see** [reply]
Are you using any kind of wireless connectivity?

**Reply by QJKX** [reply]
> To answer your question, it hops around a lot. _ping_ is between 22-45, and averages (i guess) around 30. _delay_ is between 80-120, averaging around 85? This is without skype connection, which we also use during lessons. I'm sure this drives those numbers up, but I can't check that right now. If that is useful to know lemme know and I can check next wednesday, when my lesson is planned.
> 
> Do these numbers tell you anything?

It tells me you have a network issue.
That doesn't mean you don't have a driver issue as well, but I'd ignore that until the network is sorted.

I think "delay" is a total that includes ping, so with ping jumping around, you would expect delay to, too.

 [blip](https://gfblip.appspot.com/) [blip on github](https://github.com/apenwarr/blip) might be worth a look to see what the network pattern looks like on a graph. The IP address shown is yours so you may not want to share that. On my wired desktop I see fairly consistent times for the blue marks but clusters for the green and no red.

**Reply by sbg-1** [reply]
Hi! 

@ann0see No, I use a usb hub to connect my laptop directly to the router with a cat6 cable (as old as it is, my laptop doesn't have a dedicated LAN port). I've tried two different ones, a cheap one from amazon and a more expensive one that requires an external power source. 

@QJKX: 
1)  I asked the other student (MAC +  same audio interface) what his ping and delay were, and they were significantly lower: 25 and 45 respectively. The strange thing is that I did test my laptop + audio setup at his house, with his internet (again LAN + connector hub), where my issues persisted (though I don't remember my ping and delay there).. Which would suggest.. that it's the connector thing-y that's causing a delay? If that's the case I have no idea how to circumvent this. I'm due for a new laptop I'm guessing next year, but what modern laptop still has a LAN port? How do you guys (gn) do it? 

2) thanks for the blip blip suggestion (and the suggestion not to share my IP!! I hope I would've had the wherewithal to not do that, but... ya know, the reminder is appreciated :D )! my green marks/ping is somewhere around 16 or 17, and almost a straight line (or more of a "bandwidth" I guess with how many green bars there are, but they are spaced out pretty evenly between 15 and 20 ms) and the blue marks are also very consistent, around 50 ms, and basically a straight line. There's no red marks (except for the DNS, but they're supposed to be there I think?). Either way, it seems to be pretty stable and good? I think? 
(Which is odd, when taking into consideration that on Jamulus, it's so much higher and jumps around a lot. This could suggest a network problem downstream, were it not that the other student doesn't have this issue (see 1) ping=25, latency=45 ))

Other:
Behringer's current suggestion is that I download a DAW with the intent to rule out a latency issue between my laptop and the interface. I've never used one before and wasn't really planning to, but I guess that's my weekend plans sorted. I also have a different interface to test out next wednesday. I'll report back on that.

Thanks again for chiming in!! I appreciate it loads!

**Reply by ann0see** [reply]
> I'm due for a new laptop I'm guessing next year, but what modern laptop still has a LAN port? How do you guys (gn) do it?

Somewhat unrelated, but I just use a docking station from my laptop's manufacturer and the latency is good.

**Reply by QJKX** [reply]
There's a way of showing a graph of CPU usage in task manager (or [process explorer](https://learn.microsoft.com/en-us/sysinternals/downloads/process-explorer))  so there's a red area and green.  The red is for kernel stuff including drivers - it shouldn't be much of the total, but if the driver/hardware is having issues it may show a lot of red.

You can look at that, and Blip and compare it doing nothing, to running Jamulus, and running the [DSL reports speedtest](https://www.dslreports.com/speedtest).


> Which would suggest.. that it's the connector thing-y that's causing a delay? 

It seems likely - I'm not 100% on it.

>If that's the case I have no idea how to circumvent this. 

A different one.
>

**Comment by dcorson-ticino-com** [comment]
Reading over all you wrote the problem seems to be the USB interface. 
The clue is here:
"all runs perfectly fine with my interface + mic + headphones. As soon as I hooked it up to my laptop again, however, I got the same delay issue, **even on his network**"
If I read it correctly the only difference is that in one case you need no USB interface, in the other you do.
If your compi has no other way to get wired Internet into the box you are stuck.
Maybe trying a different USB interface could help, but that is really a suggestion without much hope.

**Reply by QJKX** [reply]
To clarify I think you mean the USB *network* interface? which is definitely possible, especially with Realtek.
In the same way it could be a dodgy Ethernet cable.


The network can be eliminated by running and connecting to a local server - proving if the audio interface is in fact the problem or not. https://jamulus.io/wiki/Running-a-Server#servers-on-the-desktop


Other things I can think of are the power profile of the PC -setting this to "performance", may help.
Trying different USB ports.
Trying the laptop without the power supply connected (really!).

**Reply by QJKX** [reply]
>"all runs perfectly fine with my interface + mic + headphones. As soon as I hooked it up to my laptop again, however, I got the same delay issue, even on his network"

A VPN would do that too.

**Reply by sbg-1** [reply]
@dcorson-ticino-com :  Hi! As mentioned in my reply above, I have tried a different (to my knowledge _better_) usb hub to connect my laptop directly to the router. It helped marginally with the ping/delay, but it didn't meaningfully change my sound/issue. I don't really know a way around this either :,(  

@QJKX : Hi again! 
1) realtek my most beloathed. I'll try to see if I can get my mitts on a different ethernet cable, but from your blip blip test suggestion earlier I'm thinking my internet connection itself is pretty okay on it? I could be missing something though. trying never hurts

2) woof, I read through all that and have to admit this is way above my proverbial paygrade / level of understanding. I'm not really sure what exactly this means and certainly don't know how to implement it. I can ask my dad who's in IT though, maybe he can make sense of it. I'll get back to you on that!

3) Thanks for these suggestions. I'll play around with the performance setting (the slider's currently in the middle, I'll try putting it all the way to "best performance"). I think I've tried playing around with the (limited number of) different usb ports, but I don't remember exactly if that did anything, I'll try that again too. Same with the laptop without the power supply, although my battery is old and "doing its best" so to speak (ie hanging on by a thread), so with the behringer interface connected and having phantom power on for the mic i'm guessing my lessons will be very short indeed ;) the usb hub also has a power connector that my laptop may try to use, but it's worth a shot at least.

This troubleshooting business is no joke, huh! Thanks again everyone! you have no way of seeing it, but I'm saluting you all in gratitude!

**Reply by QJKX** [reply]
Are the fans working and not blocked?

Is the RAM OK?  (Memtest86+)

You could try running a "live" linux from a USB device https://ubuntu.com/tutorials/create-a-usb-stick-on-windows .

**Comment by sbg-1** [comment]
Hi, lads (gn), I'm back with new findings!!

Sorry for the delay in my replies, I've tried a bunch of things the past few days and wanted to collect all my "data" and compile it in a single post. Here it goes:

1) So first thing is that I borrowed a focusrite solo interface. As I suspected the latency issue persisted. I also played around with different usb ports and headphones and a different hub, and none of it made any difference. 

2) The next thing I tried has me completely mystified:
I asked my friend to bring his macbook over to my place and set it up with my equipment (audio interface + mic + headphones + ethernet hub). And what do you know? Same latency issue. Okay, so maybe it is a network issue after all, right? Except, when I tried my laptop on _**his**_ network (where he experiences no issues with jamulus) I still had the same delay. ???????  

2a) It  doubt it’s the LAN hub because I've tried 3 different ones atp, and my connection over the ethernet cable seems fine-ish (def slower than my friend's, but he has a fibre optic connection so that's to be expected. but again, I tried his internet and ran into the same issue). 
(for reference/completeness' sake: my ping is 27-35, my friend's doesn't go over 30, my delay is 65 to 79, his is between 40 and 60, my audio stream rate is 480 kbps, his is 729). I do at some point want to go over to his house with my set up again and try his internet with an ethernet to usb cable to circumvent the hub, but I’m not sure when I’ll be able to (due to traveling the coming month). 

2b) in complete desperation I've also tried manually turning the wifi completely off when connected to ethernet. I think strangely the ping and delay go down a little bit, but the latency is still there ( in case it is useful information my wifi network is a mesh of 2 wifi channels, one on 5GHz and one on 2,4GHz ) 


Other stuff i've tried:

3) DAW: I've downloaded reaper and created a track and listened to the live monitoring. that seems fine, minimal delay and totally workable. I haven't recorded anything yet though. I suppose this suggests (rules out?) that the latency between the interface and my pc is not actually the problem. 


@QJKX 

Thanks so much for these suggestions, i've run through them:

> Other things I can think of are the power profile of the PC -setting this to "performance", may help.

This did not appear to do anything unfortunately

> Trying different USB ports.

no change

> Trying the laptop without the power supply connected (really!).

This made my pc or jamulus (i'm not sure which) freak out and completely cut out my audio, so no luck with this

> Are the fans working and not blocked?

Yeah, fans are fine

> Is the RAM OK? (Memtest86+)
I didn't use memtest, just the graphs on the performance tab option on the task manager. I think it’s okay? In use: 5GB, available 2,8GB and seems stable. 

> There's a way of showing a graph of CPU usage in task manager (or [process explorer](https://learn.microsoft.com/en-us/sysinternals/downloads/process-explorer)) so there's a red area and green. The red is for kernel stuff including drivers - it shouldn't be much of the total, but if the driver/hardware is having issues it may show a lot of red.

CPU is at 7% running just jamulus, and at 57% using jamulus and skype concurrently. I don’t see any red anywhere. (I hope these are the right parameters to mention, if not, lemme know!)

> You could try running a "live" linux from a USB device https://ubuntu.com/tutorials/create-a-usb-stick-on-windows .

I have found a someone who could do this for me, but given the issue the mac had at my place, I suspect linux would run into the same underlying issue. I'm not opposed to trying this out down the line, but it would take me 6-8 weeks to get to this (as mentioned, I'll be traveling for the next couple of weeks, though I'm taking my pc and audio equipment with me so i can try a different network and keep trying other things)


If any of you are still with me and can make sense of what I've found, or have other suggestions I'd love to hear it! Hope you are all well, and as always thanks so much for all the help so far!
